### PR TITLE
#2353 Avoid using Inventory Stock Index for default stock

### PR DESCRIPTION
**Pull request by konarshankar07**
### Fixed Issues (if relevant)

1. magento/inventory#2353: Avoid using Inventory Stock Index for default stock

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/inventory?pullRequest=2505) <br/>All committers have signed the CLA.

File: app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddIsInStockFilterToCollectionPlugin.php [hunk 1]
```diff
@@ -53,7 +58,12 @@ public function aroundAddIsInStockFilterToCollection(
         $collection
     ) {
         $stockId = $this->getStockIdForCurrentWebsite->execute();
-        $this->addIsInStockFilterToCollection->execute($collection, $stockId);
+        if ($this->defaultStockProvider->getId() === $stockId) {
+            $proceed($collection);
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddIsInStockFilterToCollectionPlugin.php]: Should it be returned?

File: app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddIsInStockFilterToCollectionPlugin.php [hunk 2]
```diff
@@ -33,10 +35,13 @@ class AdaptAddIsInStockFilterToCollectionPlugin
      */
     public function __construct(
         GetStockIdForCurrentWebsite $getStockIdForCurrentWebsite,
-        AddIsInStockFilterToCollection $addIsInStockFilterToCollection
+        AddIsInStockFilterToCollection $addIsInStockFilterToCollection,
+        DefaultStockProviderInterface $defaultStockProvider = null
     ) {
         $this->getStockIdForCurrentWebsite = $getStockIdForCurrentWebsite;
         $this->addIsInStockFilterToCollection = $addIsInStockFilterToCollection;
+        $this->defaultStockProvider = $defaultStockProvider ?: ObjectManager::getInstance()
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddIsInStockFilterToCollectionPlugin.php]: This class is a plugin and therefore is not covered by the backward compatibility development policies. There is no need to inject additional dependencies this way

File: app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddStockStatusToSelectPlugin.php [hunk 3]
```diff
@@ -30,16 +32,24 @@ class AdaptAddStockStatusToSelectPlugin
...
      */
     public function __construct(
         StockResolverInterface $stockResolver,
-        AddStockStatusToSelect $addStockStatusToSelect
+        AddStockStatusToSelect $addStockStatusToSelect,
+        DefaultStockProviderInterface $defaultStockProvider = null
     ) {
         $this->stockResolver = $stockResolver;
         $this->addStockStatusToSelect = $addStockStatusToSelect;
+        $this->defaultStockProvider = $defaultStockProvider ?: ObjectManager::getInstance()
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddStockStatusToSelectPlugin.php]: Same as above

**Review by ishakhsuvarov — CHANGES_REQUESTED** [review]
* Please change the dependency injection logic as plugins are not covered by the backward compatibility policy
* Please take a look at the automatic test failures

File: app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddIsInStockFilterToCollectionPlugin.php (hunk 1 above)
Comment (konarshankar07) [reply app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddIsInStockFilterToCollectionPlugin.php]: Overridden function is not returning any thing for above class but I agree to returned in `AdaptAddStockStatusToSelectPlugin` and `AdaptAddIsInStockFilterToCollectionPlugin` classes

File: app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddIsInStockFilterToCollectionPlugin.php [hunk 4]
```diff
@@ -11,9 +11,10 @@
 use Magento\CatalogInventory\Model\ResourceModel\Stock\Status;
 use Magento\InventoryCatalog\Model\GetStockIdForCurrentWebsite;
 use Magento\InventoryCatalog\Model\ResourceModel\AddIsInStockFilterToCollection;
+use Magento\InventoryCatalogApi\Api\DefaultStockProviderInterface;
 
 /**
- * Adapt adding is in stock filter to collection for multi stocks.
+ * Class AdaptAddIsInStockFilterToCollectionPlugin
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddIsInStockFilterToCollectionPlugin.php]:
```suggestion
 * Adapt adding is in stock filter to collection for multi stocks.
```

It is the best practice to keep comments descriptive and meaningful. Previous comment seemed to provide more context on this class.

File: app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddStockStatusToSelectPlugin.php [hunk 5]
```diff
@@ -11,12 +11,13 @@
...
 use Magento\Framework\Exception\LocalizedException;
 use Magento\InventoryCatalog\Model\ResourceModel\AddStockStatusToSelect;
+use Magento\InventoryCatalogApi\Api\DefaultStockProviderInterface;
 use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
 use Magento\InventorySalesApi\Api\StockResolverInterface;
 use Magento\Store\Model\Website;
 
 /**
- * Adapt adding stock status to select for multi stocks.
+ * Class AdaptAddStockStatusToSelectPlugin
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddStockStatusToSelectPlugin.php]:
```suggestion
 * Adapt adding stock status to select for multi stocks.
```

Same as above

File: app/code/Magento/InventoryCatalogSearch/Plugin/Model/Search/FilterMapper/TermDropdownStrategy/AdaptApplyStockConditionToSelectPlugin.php [hunk 6]
```diff
@@ -10,13 +10,14 @@
...
 use Magento\Framework\DB\Select;
+use Magento\InventoryCatalogApi\Api\DefaultStockProviderInterface;
 use Magento\InventoryIndexer\Model\StockIndexTableNameResolverInterface;
 use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
 use Magento\InventorySalesApi\Api\StockResolverInterface;
 use Magento\Store\Model\StoreManagerInterface;
 
 /**
- * Adapt apply stock condition to multi stocks
+ * Class AdaptApplyStockConditionToSelectPlugin
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryCatalogSearch/Plugin/Model/Search/FilterMapper/TermDropdownStrategy/AdaptApplyStockConditionToSelectPlugin.php]:
```suggestion
 * Adapt apply stock condition to multi stocks
```

Same

File: app/code/Magento/InventoryCatalogSearch/Test/Integration/Model/Search/FilterMapper/TermDropdownStrategy/ApplyStockConditionToSelectOnDefaultStockTest.php [hunk 7]
```diff
@@ -76,12 +76,7 @@ public function testExecute()
         $this->applyStockConditionToSelect->execute('eav_index', 'eav_index_stock', $select);
 
         $result = $select->query()->fetchAll();
-
-        /**
-         * Taking into account that `inventory_stock_1` is a MySQL View referring cataloginventory_stock_status table
-         * all the legacy StockItems will get into the Default Index
-         */
-        self::assertEquals(6, count($result));
+        self::assertEquals(4, count($result));
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryCatalogSearch/Test/Integration/Model/Search/FilterMapper/TermDropdownStrategy/ApplyStockConditionToSelectOnDefaultStockTest.php:79]: Why test result changed after these code changes?

**Review by ishakhsuvarov — CHANGES_REQUESTED** [review]
* Please update comments
* Please check carefully why result has changed for the integration test. I would expect it to remain the same

File: app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddIsInStockFilterToCollectionPlugin.php (hunk 4 above)
Comment (konarshankar07) [reply app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddIsInStockFilterToCollectionPlugin.php]: As per new docblock, every function should have a short description that's why I moved this comment [here](https://github.com/magento/inventory/blob/dcc19504b5d1548e94ae283fb62868619c23e457/app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddIsInStockFilterToCollectionPlugin.php#L47) just above the function

File: app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddStockStatusToSelectPlugin.php (hunk 5 above)
Comment (konarshankar07) [reply app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddStockStatusToSelectPlugin.php]: The same thing was done here as above

File: app/code/Magento/InventoryCatalogSearch/Plugin/Model/Search/FilterMapper/TermDropdownStrategy/AdaptApplyStockConditionToSelectPlugin.php (hunk 6 above)
Comment (konarshankar07) [reply app/code/Magento/InventoryCatalogSearch/Plugin/Model/Search/FilterMapper/TermDropdownStrategy/AdaptApplyStockConditionToSelectPlugin.php]: The same thing was done here as above

File: app/code/Magento/InventoryCatalogSearch/Test/Integration/Model/Search/FilterMapper/TermDropdownStrategy/ApplyStockConditionToSelectOnDefaultStockTest.php (hunk 7 above)
Comment (konarshankar07) [reply app/code/Magento/InventoryCatalogSearch/Test/Integration/Model/Search/FilterMapper/TermDropdownStrategy/ApplyStockConditionToSelectOnDefaultStockTest.php:79]: This test is regarding the default stock test but now we are avoiding the inventory stock for the default stock so the test result should not take inventory stock index

File: app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddIsInStockFilterToCollectionPlugin.php (hunk 4 above)
Comment (ishakhsuvarov) [reply app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddIsInStockFilterToCollectionPlugin.php]: Same applies to classes, not only methods. It's ok to make class and method doc block similar, but avoiding meaningless comments

**Review by ishakhsuvarov — APPROVED** [review]

**Pull request merged**